src/js/common/utils/lodash-replacements/deconstructPath.js

```diff
@@ -0,0 +1,36 @@
+const arrayIndexRE = /(.+)\[(\d+)\]$/;
+const numOnlyRE = /^(\d+)$/;
+
+
+/**
+ * Takes a string and casts it into an array.
+ * Can take strings like a.b[4].c
+ *
+ * @param {string} path
+ * @return {Array}
+ */
+export default function deconstructPath(path) {
+  const arrayPath = path.split('.');
```

Comment:
Since array/property indexes will coerce to the right type, I think you could also do `.split(/[\.\[\]]/)` and filter out empty strings.

Reply:
Ahhh, instead of doing a regex match later? Clever. I like it!